abci/server/socket_server.go

```diff
@@ -186,61 +195,100 @@ func (s *SocketServer) handleRequests(closeConn chan error, conn io.Reader, resp
 		}
 		s.appMtx.Lock()
 		count++
-		s.handleRequest(req, responses)
+		resp, err := s.handleRequest(context.TODO(), req)
```

Comment:
This, and the few hunks below, seems to me not needed to backport the initial `FinalizeBlock` patch.
Hopefully, it won't create conflicts when backporting further patches.